### ISSUE TITLE
DeepLift and DeepLiftShap model requirements potentially undocumented

### ISSUE THREAD

**Issue by wbeardall** [pytorch/captum#1177]
I am attempting to use DeepLift and/or DeepLiftShap to calculate attribution values for a particular model. This model takes an arbitrary tuple of tensors as input, and returns a single tensor output. Whilst I can calculate attributions fine with, e.g. `KernelShap`, attempting to utilise `DeepLift` or `DeepLiftShap` breaks the model forward pass due to the `captum.attr.DeepLift._construct_forward_func` method. 

This method, when called in `captum.attr.DeepLift.attribute`, passes both the inputs and baselines to the model's `__call__` method as a tuple: 

```
wrapped_forward_func = self._construct_forward_func(
  self.model,
  (inputs, baselines),
  ...)
```

However, the input Pytorch module is not designed to have two tuples of tuples passed to its `__call__` method, nor is it designed to stack the resulting outputs along dim 1, as is apparently needed for the call to `_select_targets` in the following line of `_construct_forward_func`:

```
_select_targets(
  torch.cat((model_out[:, 0], model_out[:, 1])), target
)
```

Is there something that I have missed here, or should the docs be updated to illustrate the required wrapping of the Pytorch module?
